### PR TITLE
Log reader proxy

### PR DESCRIPTION
**Pull request by andrzejgorski**

### CODE REVIEW

File: whylog_vim/output_formater/output_aggregator.py [hunk 1]
```diff
@@ -20,11 +20,17 @@ def create_button(self, callback_function, line=None):
...
+    def set_default_callback_function(self, default_callback_function):
+        self.default_callback_function = default_callback_function
+
     def call_button(self, line_number):
         try:
             self.buttons[line_number]()
         except KeyError:
-            pass
+            try:
+                self.default_callback_function()
```
Comment (kgromadzki) [inline whylog_vim/output_formater/output_aggregator.py:31]: I didn't find definition of default_callback_function. I guess that will be implemented soon. It explain pass in except block. Create mock for unimplemented method.

File: whylog_vim/proxy/log_reader/__init__.py [hunk 2]
```diff
@@ -1,10 +1,25 @@
...
     def __init__(self, editor, log_reader):
         self.editor = editor
-        self.log_reader = log_reader
+        self._log_reader = log_reader
 
     def new_query(self):
-        pass
+        front_input = self.editor.get_front_input()
+        query_output = self._log_reader.get_causes(front_input)
+
```
Comment (kgromadzki) [inline whylog_vim/proxy/log_reader/__init__.py]: Remove blank line

File: whylog_vim/tests/tests_proxy/tests_unit_proxy_main.py [hunk 3]
```diff
@@ -1,25 +1,34 @@
...
 class UnitTestWhylogProxy(TestCase):
-    def tests_unit_check_log_reader_states_of_whylog_proxy(self):
-        whylog_proxy = WhylogProxy(MagicMock())
+    def setUp(self):
+        mock_editor = MagicMock()
+        mock_editor.get_front_input.return_value = FrontInput(
+            1, 'line content', LineSource(
+                'host', 'path'
+            )
+        )
```
Comment (kgromadzki) [inline whylog_vim/tests/tests_proxy/tests_unit_proxy_main.py]: Add # yapf: disable

File: whylog_vim/output_formater/output_aggregator.py (hunk 1 above)
Comment (andrzejgorski) [inline whylog_vim/output_formater/output_aggregator.py:31]: Look at line 23.

File: whylog_vim/tests/tests_proxy/tests_unit_proxy_main.py [hunk 4]
```diff
@@ -1,25 +1,31 @@
...
 from whylog_vim.proxy import WhylogProxy
 
 
 class UnitTestWhylogProxy(TestCase):
-    def tests_unit_check_log_reader_states_of_whylog_proxy(self):
-        whylog_proxy = WhylogProxy(MagicMock())
+    def setUp(self):
+        mock_editor = MagicMock()
+        mock_editor.get_front_input.return_value = FrontInput(  # yapf:disable
+                1, 'line content', LineSource('host', 'path'))
```
Comment (kgromadzki) [inline whylog_vim/tests/tests_proxy/tests_unit_proxy_main.py]: Move yapf disable to this line

File: whylog_vim/output_formater/output_aggregator.py (hunk 1 above)
Comment (kgromadzki) [inline whylog_vim/output_formater/output_aggregator.py:31]: Okay, I understood. But you still have a pass in except clause.

File: whylog_vim/tests/tests_proxy/tests_unit_proxy_main.py [hunk 5]
```diff
@@ -1,25 +1,31 @@
...
 from whylog_vim.proxy import WhylogProxy
 
 
 class UnitTestWhylogProxy(TestCase):
-    def tests_unit_check_log_reader_states_of_whylog_proxy(self):
-        whylog_proxy = WhylogProxy(MagicMock())
+    def setUp(self):
+        mock_editor = MagicMock()
+        mock_editor.get_front_input.return_value = FrontInput(
+                1, 'line content', LineSource('host', 'path'))  # yapf:disable
```
Comment (ppolewicz) [inline whylog_vim/tests/tests_proxy/tests_unit_proxy_main.py:14]: we have facebook style enabled so this is improper formatting. Do not disable yapf here before formatting the code properly.

**Pull request merged**